pkg/localcontroller/managers/incrementallearning/incrementallearningjob.go

```diff
@@ -383,13 +389,108 @@ func (im *Manager) Delete(message *clienttypes.Message) error {
 	return nil
 }
 
+// updateJobFromDB updates job from db
+func (im *Manager) updateJobFromDB(job *Job) error {
+	var err error
+
+	previousJob, err := db.GetResource(job.JobConfig.UniqueIdentifier)
+	if err != nil {
+		return err
+	}
+
+	m := metav1.ObjectMeta{}
+	if err != json.Unmarshal([]byte(previousJob.ObjectMeta), &m) {
+		return err
+	}
+
+	rounds, ok := m.Annotations[AnnotationsRoundsKey]
+	if !ok {
+		return nil
+	}
+
+	if job.JobConfig.Rounds, err = strconv.Atoi(rounds); err != nil {
+		return err
+	}
+
+	numberOfSamples, ok := m.Annotations[AnnotationsNumberOfSamplesKey]
+	if !ok {
+		return nil
+	}
+
+	if job.JobConfig.DataSamples.PreviousNumbers, err = strconv.Atoi(numberOfSamples); err != nil {
+		return err
+	}
+
+	dataFileOfEval, ok := m.Annotations[AnnotationsDataFileOfEvalKey]
+	if !ok {
+		return nil
+	}
+
+	localURL, err := job.JobConfig.Storage.Download(dataFileOfEval, "")
+
+	if !job.JobConfig.Storage.IsLocalStorage {
+		defer os.RemoveAll(localURL)
+	}
+
+	if err != nil {
+		return err
+	}
+
+	samples, err := dataset.GetSamples(dataFileOfEval)
+	if err != nil {
+		klog.Errorf("read file %s failed, error: %v", dataFileOfEval, err)
+		return err
+	}
+
+	job.JobConfig.DataSamples.EvalVersionSamples = append(job.JobConfig.DataSamples.EvalVersionSamples, samples)
+
+	return nil
+}
+
+// updateAnnotations updates job annotations
+func (im *Manager) updateJobToDB(job *Job) error {
+	ann := job.ObjectMeta.Annotations
+	if ann == nil {
+		ann = make(map[string]string)
+	}
+
+	ann[AnnotationsRoundsKey] = strconv.Itoa(job.JobConfig.Rounds)
+	ann[AnnotationsNumberOfSamplesKey] = strconv.Itoa(job.JobConfig.DataSamples.PreviousNumbers)
+	ann[AnnotationsDataFileOfEvalKey] = job.JobConfig.EvalDataURL
+
+	if err := db.SaveResource(job.JobConfig.UniqueIdentifier, job.TypeMeta, job.ObjectMeta, job.Spec); err != nil {
+		return err
+	}
+
+	return nil
+}
+
 // initJob inits the job object
-func (im *Manager) initJob(job *Job) error {
+func (im *Manager) initJob(job *Job, name string) error {
+	job.JobConfig = new(JobConfig)
+
 	jobConfig := job.JobConfig
+	jobConfig.UniqueIdentifier = name
+
+	jobConfig.Storage = storage.Storage{IsLocalStorage: false}
+	credential := job.ObjectMeta.Annotations[runtime.SecretAnnotationKey]
+	if credential != "" {
+		if err := job.JobConfig.Storage.SetCredential(credential); err != nil {
+			return fmt.Errorf("failed to set job(name=%s)'s storage credential, error: %+v", name, err)
```

Comment:
```suggestion
			return fmt.Errorf("job(name=%s) failed to set storage credential: %w", name, err)
```

Reply:
1. no job name in returned error.
2. wrap error using %w.

Reply:
fix it